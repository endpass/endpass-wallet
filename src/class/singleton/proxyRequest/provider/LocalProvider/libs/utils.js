import regPath from 'path-to-regexp';

const utils = {
  getRouteKey(routesRegexp, routePath) {
    return Object.keys(routesRegexp).find(key => {
      const re = routesRegexp[key];
      return re.test(routePath);
    });
  },
  getPath(serverUrl, url) {
    const levels = url.split(serverUrl);
    const keysPath = levels.pop() || '';
    const ret = keysPath[0] === '/' ? keysPath.substring(1) : keysPath;

    return ret;
  },
  getPathArgs(routesRegexp, key, routePath) {
    const keys = regPath.parse(key);
    const re = routesRegexp[key];
    re.lastIndex = 0; // drop exec position

    const values = re.exec(routePath);
    return keys.reduce((map, item, index) => {
      if (item && item.name) {
        map[item.name] = values[index];
      }
      return map;
    }, {});
  },
};

export default utils;
