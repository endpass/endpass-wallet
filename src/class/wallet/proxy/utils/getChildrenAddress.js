export default function(hdWallet, offset, limit) {
  return [...Array(limit)].map((_, i) =>
    hdWallet
      .deriveChild(offset + i)
      .getWallet()
      .getChecksumAddressString(),
  );
}
