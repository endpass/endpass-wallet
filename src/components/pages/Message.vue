<template>
  <base-page>
    <template slot="title">Message</template>

    <div class="tabs">
      <ul>
        <li :class="[ isSignTabActive ? 'is-active' : '']">
          <a
            @click="setSignTabActive"
            data-test="sign-message-tab"
          >
            Sign
          </a>
        </li>
        <li :class="[ isVerifyTabActive ? 'is-active' : '']">
          <a
            @click="setVerifyTabActive"
            data-test="verify-message-tab"
          >
            Verify
          </a>
        </li>
      </ul>
    </div>
    <sign-message v-show="isSignTabActive"/>
    <verify-message v-show="isVerifyTabActive"/>
  </base-page>
</template>

<script>
import BasePage from '@/components/pages/Base';
import SignMessage from '@/components/SignMessage';
import VerifyMessage from '@/components/VerifyMessage';
import privatePage from '@/mixins/privatePage';

const ACTION_TYPES = {
  SIGN: 'SIGN',
  VERIFY: 'VERIFY',
};

export default {
  name: 'MessagePage',
  data: () => ({
    action: ACTION_TYPES.SIGN,
  }),
  computed: {
    isSignTabActive() {
      return this.action === ACTION_TYPES.SIGN;
    },
    isVerifyTabActive() {
      return this.action === ACTION_TYPES.VERIFY;
    },
  },
  methods: {
    setSignTabActive() {
      this.action = ACTION_TYPES.SIGN;
    },
    setVerifyTabActive() {
      this.action = ACTION_TYPES.VERIFY;
    },
  },

  mixins: [privatePage],

  components: {
    BasePage,
    SignMessage,
    VerifyMessage,
  },
};
</script>
