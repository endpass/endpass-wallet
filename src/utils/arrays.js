export const uniq = arr =>
  arr.reduce((acc, item) => (acc.includes(item) ? acc : acc.concat(item)), []);

export const mapArrayByProp = (arr, prop) =>
  arr.reduce((acc, item) => {
    const target = item[prop];

    if (target) {
      return Object.assign(acc, {
        [target]: item,
      });
    }

    return acc;
  }, {});

export default {
  uniq,
  mapArrayByProp,
};
