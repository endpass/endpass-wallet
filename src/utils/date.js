import dayjs from 'dayjs';
import relativeTime from 'dayjs/plugin/relativeTime';

/**
 * Setting up dayjs globally
 */
dayjs.extend(relativeTime);

export const formateDate = date => dayjs(date).format('YYYY-MM-DD H:mm');

export const fromNow = date => dayjs(date).fromNow();

export default {
  formateDate,
  fromNow,
};
