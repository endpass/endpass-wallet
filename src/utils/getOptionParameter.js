export default function getOptionParameter(item, value) {
  return item instanceof Object ? item[value] : item;
}
