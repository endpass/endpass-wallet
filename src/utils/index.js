import store from '../store';

export { default as http } from './http';

export const kebabToCamel = input =>
  input.replace(/-([a-z])/g, g => g[1].toUpperCase());

export const getInitializedValueFromStore = (module, field) => {
  return new Promise(resolve => {
    const value = module[field];

    if (value === null) {
      const unwatch = store.watch(
        () => module[field],
        newValue => {
          unwatch();
          resolve(newValue);
        },
      );
    } else {
      resolve(value);
    }
  });
};
