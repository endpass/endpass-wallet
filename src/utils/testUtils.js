import { kebabCase } from 'lodash';
import { getFrom } from '@/utils/objects';

function convertDirectivesToAttrs(directives) {
  const attrs = {};

  if (directives) {
    directives.forEach(({ rawName, expression }) => {
      attrs[rawName] = expression;
    });
  }

  return attrs;
}

function convertListenersToAttrs(listeners) {
  const attrs = {};

  if (listeners) {
    Object.keys(listeners).forEach(key => {
      const handlers = [].concat(listeners[key]);
      const handler = handlers.find(h => {
        const name = getFrom(h, 'fns.name', 'name') || '';

        return name.includes('bound ');
      });

      if (handler) {
        attrs[`v-on:${key}`] = (handler.fns || handler).name;
      }
    });
  }

  return attrs;
}

function convertModelToAttrs(model) {
  return model ? { 'v-model': model.expression } : {};
}

function convertSlotToAttrs(slot) {
  return slot ? { slot } : {};
}

function generateElement(createElement, vnode) {
  return vnode.componentOptions
    ? generateElementFromComponent(createElement, vnode)
    : generateElementFromHTML(createElement, vnode);
}

function getChildrenElements(createElement, children) {
  return children
    ? children.map(child => generateElement(createElement, child))
    : undefined;
}

function generateElementFromComponent(
  createElement,
  { data, componentOptions },
) {
  const { attrs, directives, model } = data;
  const { tag, listeners, children } = componentOptions;
  const componentData = {
    ...data,
    attrs: {
      ...attrs,
      ...convertDirectivesToAttrs(directives),
      ...convertListenersToAttrs(listeners),
      ...convertModelToAttrs(model),
      class: data.staticClass,
    },
    hook: undefined,
    staticClass: undefined,
  };

  if (listeners) {
    componentData.on = Object.assign({}, listeners);
  }

  return createElement(
    tag,
    componentData,
    getChildrenElements(createElement, children),
  );
}

function generateElementFromHTML(createElement, vnode) {
  if (!vnode.data) {
    return vnode;
  }

  const { tag, data, children } = vnode;
  const { directives, on, model, attrs = {}, slot } = data;

  return createElement(
    tag,
    {
      ...data,
      attrs: {
        ...attrs,
        ...convertDirectivesToAttrs(directives),
        ...convertListenersToAttrs(on),
        ...convertModelToAttrs(model),
        ...convertSlotToAttrs(slot),
      },
    },
    getChildrenElements(createElement, children),
  );
}

export function generateStubs(Component) {
  return Object.values(Component.components).reduce((stubs, stubComponent) => {
    let elementName;

    if (stubComponent.name) {
      elementName = kebabCase(stubComponent.name);
    }

    return Object.assign(stubs, {
      [elementName]: {
        render(createElement) {
          return generateElementFromComponent(createElement, this.$vnode);
        },
      },
    });
  }, {});
}

export default {
  generateStubs,
};
