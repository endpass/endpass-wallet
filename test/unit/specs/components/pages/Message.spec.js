import Vuex from 'vuex';
import { shallow, createLocalVue } from '@vue/test-utils';
import Message from '@/components/pages/Message';
import { generateStubs } from '@/utils/testUtils';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('Message page', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        accounts: {
          namespaced: true,
          state: {
            wallet: {
              isPublic: false,
            },
          },
          getters: {
            isPublicAccount: () => true,
          },
        },
      },
    });
    wrapper = shallow(Message, {
      stubs: generateStubs(Message),
      localVue,
      store,
    });
  });

  describe('render', () => {
    it('should render the initial state of the page', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('should set Sign tab', () => {
      wrapper.setData({
        action: 'SIGN',
      });

      expect(wrapper.element).toMatchSnapshot();
    });

    it('should set Verify tab', () => {
      wrapper.setData({
        action: 'VERIFY',
      });

      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
