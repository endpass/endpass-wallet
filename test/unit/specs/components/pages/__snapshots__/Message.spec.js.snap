// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message page render should render the initial state of the page 1`] = `
<base-page>
  <template
    slot="title"
  >
    Message
  </template>
   
  <div
    class="tabs"
  >
    <ul>
      <li
        class="is-active"
      >
        <a>
          Sign
        </a>
      </li>
       
      <li
        class=""
      >
        <a>
          Verify
        </a>
      </li>
    </ul>
  </div>
   
  <sign-message
    v-show="isSignTabActive"
  />
   
  <verify-message
    style="display: none;"
    v-show="isVerifyTabActive"
  />
</base-page>
`;

exports[`Message page render should set Sign tab 1`] = `
<base-page>
  <template
    slot="title"
  >
    Message
  </template>
   
  <div
    class="tabs"
  >
    <ul>
      <li
        class="is-active"
      >
        <a>
          Sign
        </a>
      </li>
       
      <li
        class=""
      >
        <a>
          Verify
        </a>
      </li>
    </ul>
  </div>
   
  <sign-message
    v-show="isSignTabActive"
  />
   
  <verify-message
    style="display: none;"
    v-show="isVerifyTabActive"
  />
</base-page>
`;

exports[`Message page render should set Verify tab 1`] = `
<base-page>
  <template
    slot="title"
  >
    Message
  </template>
   
  <div
    class="tabs"
  >
    <ul>
      <li
        class=""
      >
        <a>
          Sign
        </a>
      </li>
       
      <li
        class="is-active"
      >
        <a>
          Verify
        </a>
      </li>
    </ul>
  </div>
   
  <sign-message
    style="display: none;"
    v-show="isSignTabActive"
  />
   
  <verify-message
    style=""
    v-show="isVerifyTabActive"
  />
</base-page>
`;
